protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -584,6 +590,48 @@ definitions:
       - overwrite
       #- upsert_dedup # TODO chris: SCD Type 1 can be implemented later
       - append_dedup # SCD Type 1 & 2
+      - insert
+      - update
+      - upsert
+      - soft_delete
+  DestinationCatalog:
+    type: object
+    additionalProperties: true
+    description: Catalog for a destination
+    required:
+      - operations
+    properties:
+      operations:
+        description: An array of operations that can be performed on destination objects.
+        type: array
+        items:
+          "$ref": "#/definitions/DestinationOperation"
+  DestinationOperation:
+    type: object
+    additionalProperties: true
+    description: A possible operation on a destination object.
+    required:
+      - object_name
+      - operation
+      - json_schema
+    properties:
+      object_name:
+        description: The name of the destination object.
+        type: string
+      operation:
+        description: The operation to be performed on the destination object.
+        "$ref": "#/definitions/DestinationSyncMode"
+      json_schema:
+        description: Stream schema using Json Schema specs.
+        type: object
+        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+      record_matchers:
+        description: Keys on which we can map to the destination object. Those keys can be composite.
```

Comment:
@pedroslopez do you recall what is meant by "composite" here? Are there destinations where multiple record_matchers are used?

I'm assuming the array of arrays is so that we can do something like:

```
record_matchers: [
 ['field_one'],
 ['field_two', 'nested_property']
]
```

Where either `field_one` or `field_two.nested_property` is the matcher.

Does that sound right? If so, I will just slightly adjust the description here.

Reply:
One more question I have: do we know if any destinations require us to specify _which_ record_matcher we have chosen? Or is it just implicitly understood based on what we pass in? (maybe no one knows the answer to this yet)

Reply:
Talked with @pedroslopez: 
- we don't think composite keys are required. We have not found a use case for one yet.
- in the configured catalog, the primary key must be set to one of the record_matchers (the dropdown for "upsert id" is essentially the list of record_matchers)
  - what do we do for source defined primary keys? can we overwrite that somehow?
- another naming idea: matching_keys, supported_matching_keys (it's kind of confusing that "keys" is not present in the name record_matchers, despite it being mapped to the primary_key field)